packages/react-components/react-tree/library/src/utils/ImmutableMap.ts

```diff
@@ -1,77 +1,98 @@
-export interface ImmutableMap<Key, Value> {
-  clear(): ImmutableMap<Key, Value>;
-  delete(key: Key): ImmutableMap<Key, Value>;
-  /**
-   * Returns a specified element from the Map object. If the value that is associated to the provided key is an object, then you will get a reference to that object and any change made to that object will effectively modify it inside the Map.
-   * @returns Returns the element associated with the specified key. If no element is associated with the specified key, undefined is returned.
-   */
-  get(key: Key): Value | undefined;
-  /**
-   * @returns boolean indicating whether an element with the specified key exists or not.
-   */
-  has(key: Key): boolean;
-  /**
-   * Adds a new element with a specified key and value to the Map. If an element with the same key already exists, the element will be updated.
-   */
-  set(key: Key, value: Value): ImmutableMap<Key, Value>;
+const internalMapSymbol = Symbol('#internalMap');
+
+export class ImmutableMap<Key, Value> implements Iterable<[Key, Value]> {
+  public static empty: ImmutableMap<never, never> = new ImmutableMap(new Map<never, never>());
```

Comment:
<img width="919" alt="image" src="https://github.com/user-attachments/assets/49308776-6d73-4b37-8b10-9dd1361800e1">

Why `empty` is needed then?

Reply:
Good question. I guess it's redundant, I'll follow up